motionplan/motionPlanner.go

```diff
@@ -549,8 +537,15 @@ func CheckPlan(
 		}, nil
 	}
 
+	var iterStartVal int
```

Comment:
This bit seems pretty fishy to me.  Why is this necessary

Reply:
This was introduced because we changed the length of the `plan` object which is received by `CheckPlan` as passed in through `obstaclesIntersectPlan`. 

For relative cases, I believe it makes sense to pass in a plan which includes _not only_ the remaining inputs + poses left to be executed, **but also** the most recent already executed input and reached pose. The benefit of doing this is that we can determine the value of the `formerRunningPose` without having to also calculate the values of our `pathPosition` and `lastPose`. 
Note: we care about the most recently reached pose since that is the value we must compose with `frame.Transform(currentInputs)` to place us in absolute coordinates.

Ultimately, this allows for us to rely less on what controls are telling us about the state of the robot and instead use the immutable plan object to inform us.

However! `iterStartVal` is introduced solely for `TestArmGantryCheckPlan` since this is the only code path which tests `CheckPlan` for an already absolute plan.

Reply:
I see now.  Would it be possible to change the way we express absolute plans to get rid of the need to do this case structure?

Reply:
We could have the invariant that each plan always contains the first index being where it starts at (which now that I write it I think is already true)

Reply:
This is making me wonder if we've set up `CheckPlan` poorly.

What we have right now is a public function with the undocumented requirement of "pass in only the plan left to execute, except going back in time one waypoint". Which is a weird way to capture state.

I am wondering if perhaps we want to instead pass in entire plans, and pass in the currently executing index, instead. Not necessarily in this PR.

Reply:
This would make it more apparent how this function is meant to be used I think. Coming at the cost of being a little more verbose.  Would be fine with this or else just documenting a little better

Reply:
I am happy to do this as part of my course correction work. Since we'll need to pass segments into this function for that to work properly, there will be a significant reworking of `CheckPlan` as part of that.

Reply:
> I am wondering if perhaps we want to instead pass in entire plans, and pass in the currently executing index, instead. Not necessarily in this PR.

I think this is a good way to solve this issue. I will incorporate this change and push it up.